tree-walk/src/parser.cpp

```diff
@@ -257,6 +280,36 @@ std::unique_ptr<stmt::Stmt> blockStatement(TokenStream &tokStream) {
   throw ParseException("Block has no closing brace.", blockStart);
 }
 
+std::unique_ptr<stmt::Stmt> classStatement(TokenStream &tokStream) {
+  int clsStart = tokStream.peek().line;
+  if (TokenType::IDENTIFIER != tokStream.peek().type) {
+    throw ParseException("class declaration not followed by identifier!",
+                         tokStream.peek().line);
+  }
+  auto cls = std::make_unique<stmt::Stmt>(stmt::Class{tokStream.peek().value});
+  tokStream.next();
+
+  if (TokenType::LEFT_BRACE != tokStream.peek().type) {
+    throw ParseException(
+        "class identifier needs to be followed by opening brace!",
+        tokStream.peek().line);
+  }
+  tokStream.next();
+
+  while (tokStream.hasNext()) {
+    if (TokenType::RIGHT_BRACE == tokStream.peek().type) {
+      tokStream.next();
+      return cls;
+    }
+    tokStream.next();
```

Comment:
I think we can get rid of this next() call and adhere to the spec where functions aren't preceded by `fun` when they're a method.